packages/camera/camera_android/android/src/main/java/io/flutter/plugins/camera/VideoRenderer.java

```diff
@@ -0,0 +1,352 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+package io.flutter.plugins.camera;
+
+import static android.os.SystemClock.uptimeMillis;
+
+import android.graphics.SurfaceTexture;
+import android.opengl.EGL14;
+import android.opengl.EGLConfig;
+import android.opengl.EGLContext;
+import android.opengl.EGLDisplay;
+import android.opengl.EGLExt;
+import android.opengl.EGLSurface;
+import android.opengl.GLES11Ext;
+import android.opengl.GLES20;
+import android.opengl.GLUtils;
+import android.opengl.Matrix;
+import android.os.Handler;
+import android.os.HandlerThread;
+import android.util.Log;
+import android.view.Surface;
+import java.nio.ByteBuffer;
+import java.nio.ByteOrder;
+
+/** Renders video onto texture. */
```

Comment:
Please provide more context about what this class is doing and how it fits into camera overall; if something significant about code structure isn't clear to reviewers (as was the case with this class), that's an indication that there's not enough commenting in the code. PR discussion is useful, but doesn't help later readers of the code.